### PR TITLE
replace examples to random.Generator

### LINKED ISSUE #121: Make sure the `random.Generator` interface is used in all tutorials
The tutorials should use the recommended `random.Generator` interface for random number generation. A quick `grep` shows that there are some instances of the older usage in the following files:
 - tutorial-nlp-from-scratch.md
 - text_preprocessing.py

It would be a nice improvement to replace these instances with the recommended random interface.

### PR DESCRIPTION
**Pull request by willhoh**
This should fixes #121. Replace legacy Random Generator (RandomState) with the new Random Generator in the tutorial.

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
I think it makes more sense to create the random instance outside of the functions/methods so that you're not creating a new instance every time the function/method is called.

**Comment by willhoh** [comment]
I agree with you, but I don't know how to put it properly in the examples. The function 'initialise_params' stands pretty much alone, so it is difficult to create the random instance at the top of the class or already during import and to document this. If you know how to do this, I would be very happy if you give me an example and I implement it.

**Comment by rossbar** [comment]
One simple way is just to create the instance at the top level of the file, e.g. where numpy is imported:

```python
import numpy as np
rng = np.random.default_rng()
```

Then use `rng` throughout.

File: content/text_preprocessing.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import pandas as pd
 import argparse
 import numpy as np
+rng = np.random.default_rng() # crating the random instance
```
Comment (Mukulikaa) [inline content/text_preprocessing.py:4]: Just a nit: maybe include this line after all the imports?

**Review by Mukulikaa — APPROVED** [review]
LGTM, thanks @willhoh!

File: content/tutorial-nlp-from-scratch.md [hunk 2]
```diff
@@ -575,7 +578,7 @@ def forward_prop(X_vec, parameters, input_dim):
         # Retrieve word corresponding to current time step
         x = X_vec[t]
         # Retrieve the embedding for the word and reshape it to make the LSTM happy
-        xt = emb_matrix.get(x, np.random.rand(input_dim, 1))
+        xt = emb_matrix.get(x, rng.standard_normal(size=(input_dim, 1)))
```
Comment (rossbar) [inline content/tutorial-nlp-from-scratch.md:581]: This one was a `rand`, not `randn`, so I think it should be replaced with `rng.random(size=`

**Review by rossbar — CHANGES_REQUESTED** [review]
This is much better thanks @willhoh , just one translation mixup that needs to be fixed then I think it's good to go!

File: content/tutorial-nlp-from-scratch.md (hunk 2 above)
Comment (willhoh) [reply content/tutorial-nlp-from-scratch.md:581]: Yes, you are right. Should be `rng.random(input_dim, 1)`

**Review by rossbar — APPROVED** [review]
LGTM, thanks @willhoh !

**Pull request merged**